stonesoup/tracker/gaussianmixture.py

```diff
@@ -0,0 +1,213 @@
+from .base import Tracker
+from ..base import Property
+from ..reader import DetectionReader
+from ..predictor import Predictor
+from ..types import TaggedWeightedGaussianState, GaussianMixtureState, Track
+from ..updater import Updater
+from ..hypothesiser import GMMahalanobisDistanceHypothesiser
+from ..dataassociator.neighbour import GaussianMixtureAssociator
+from ..mixturereducer import GaussianMixtureReducer
+from .. import measures
+
+
+class GaussianMixtureMultiTargetTracker(Tracker):
+    """
+    Base class for Gaussian Mixture style implementations of
+    point process derived filters
+    """
+    detector = Property(
+        DetectionReader,
+        default=None,
+        doc="Detector used to generate detection objects.")
+    predictor = Property(
+        Predictor,
+        default=None,
+        doc="Predictor used to predict the objects to their new state.")
+    updater = Property(
+        Updater,
+        default=None,
+        doc="Updater used to update the objects to their new state.")
+    gaussian_mixture = Property(
+        GaussianMixtureState,
+        default=None,
+        doc="""Gaussian Mixture modelling the
+                intensity over the target state space.""")
+    target_tracks = Property(
+        dict,
+        default={},
+        doc="""Dictionary containing the unique tags as keys and target
+               tracks objects as values.""")
+    data_associator = Property(
+        GaussianMixtureAssociator,
+        default=None,
+        doc="Association algorithm to pair predictions to detections")
+    reducer = Property(
+        GaussianMixtureReducer,
+        default=None,
+        doc="Reducer used to reduce the number of components in the mixture.")
+    prob_of_detection = Property(
+        float,
+        default=0.9,
+        doc="""The probability that an exisiting target is detected
+                at each timestep""")
+    prob_of_survival = Property(
+        float,
+        default=0.999,
+        doc="""The probability that an exisiting target surivives
+                until the next timestep""")
+    clutter_spatial_density = Property(
+        float,
+        default=1e-10,
+        doc="""The clutter intensity at a point in the state point.""")
+    birth_component = Property(
+        TaggedWeightedGaussianState,
+        default=None,
+        doc="""The birth component. The weight is equal to the mean of the
+        expected number of births per timestep (Poission distributed)""")
+
+    def __init__(self, *args, **kwargs):
+        if ("association_threshold" in kwargs):
```

Comment:
So I think these extra kwargs should be removed, as these should be specified as part of the hypothesier/reduce component passed in. This will mean this tracker is more generic and able to support multiple hypothesisers/reducers.